ingestion/src/metadata/profiler/metrics/static/distinct_count.py

```diff
@@ -43,13 +42,13 @@ def metric_type(self):
     def fn(self):
         return func.count(distinct(column(self.col.name)))
 
-    def df_fn(self, df=None):
-        from pandas import DataFrame  # pylint: disable=import-outside-toplevel
-
-        df = cast(DataFrame, df)
-
+    def df_fn(self, dfs=None):
         try:
-            return len(set(df[self.col.name].values.tolist()))
+            import pandas as pd  # pylint: disable=import-outside-toplevel
+
+            return len(
+                pd.concat((df[self.col.name] for df in dfs)).dropna().drop_duplicates()
+            )
```

Comment:
A few things here:

1. let's handle memory error
```python
try:
    df = pd.concat([df[self.col.name] for df in dfs])
except MemoryError:
    ...
```
3. let's use `len(df["self.col.name"].value_counts().index)` -> this will handle `nan` and will be probably more efficient than `dropna` and then `dropduplicates` https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.value_counts.html

Reply:
added memory error.
and modified it to use value_counts